src/shared/components/main.tsx

```diff
@@ -1,12 +1,432 @@
+  CARD_GRADIENT,
+  DonateBlock,
+  TEXT_GRADIENT,
+} from "./common";
+import { InstancePicker } from "./instance-picker";
+import classNames from "classnames";
+
+interface MainProps {
+  i: Main;
+}
+
+const TitleBlock = ({ i }: MainProps) => (
+  <div className="py-16 flex flex-col items-center">
+    <div className="flex flex-col items-center gap-4 mb-8">
+      <p className={`text-6xl font-bold ${TEXT_GRADIENT}`}>Lemmy</p>
+      <p className="text-3xl font-medium text-center">{i18n.t("lemmy_desc")}</p>
+    </div>
+    <div className="flex flex-row justify-around gap-4">
+      <JoinServerButton i={i} />
+      <SeeAllServersButton />
+    </div>
+  </div>
+);
+
+const carouselImages = [
+  "/static/assets/images/main_screen_2.webp",
+  "/static/assets/images/main_screen_3.webp",
+  "/static/assets/images/main_screen_1.webp",
+];
+
+const CarouselBlock = () => (
+  <div>
+    <div className="carousel carousel-center p-8 space-x-8 rounded-box mt-16">
+      {carouselImages.map((image, i) => (
+        <div id={`item-${i}`} className="carousel-item w-9/12 lg:w-5/12">
+          <img
+            src={image}
+            className={classNames("rounded-box border-8 z-10", {
+              "border-primary/[.15]": i & 1,
+              "border-secondary/[.15]": !(i & 1),
+            })}
+          />
+        </div>
+      ))}
+    </div>
+    <div className="flex justify-center w-full py-2 gap-4">
+      {carouselImages.map((_, i) => (
+        <a href={`#item-${i}`} className={TEXT_GRADIENT}>
+          ●
+        </a>
+      ))}
+    </div>
+  </div>
+);
+
+const JoinServerButton = ({ i }: MainProps) => (
+  <button
+    className="btn btn-primary text-white normal-case z-10"
+    onClick={() => {
+      i.setState({ resetInstancePicker: true });
+      (document.getElementById("picker") as any).showModal();
+    }}
+  >
+    {i18n.t("join_a_server")}
+  </button>
+);
+
+const SeeAllServersButton = () => (
+  <Link
+    to="/instances"
+    className="btn btn-secondary text-white normal-case z-10"
+  >
+    {i18n.t("see_all_servers")}
+  </Link>
+);
+
+const FollowCommunitiesBlock = ({ i }: MainProps) => (
+  <div className="flex flex-col items-center">
+    <div className={`card card-bordered ${CARD_GRADIENT} shadow-xl`}>
+      <div className="card-body items-center px-8 md:px-32 py-16">
+        <T
+          i18nKey="follow_communities"
+          className="card-title font-bold text-4xl text-center mb-3 inline-block"
+        >
+          #<span className={TEXT_GRADIENT}>#</span>
+        </T>
+        <p className="text-sm text-gray-300 text-center mb-6">
+          {i18n.t("lemmy_long_desc")}
+        </p>
+        <JoinServerButton i={i} />
+      </div>
+    </div>
+  </div>
+);
+
+const FeatureCard = ({ pic, title, subtitle, classes }) => (
+  <div className={`card ${CARD_GRADIENT} shadow-xl ${classes}`}>
+    <figure className="p-4">
+      <img src={pic} className="rounded-xl w-full object-fill min-h-[300px]" />
+    </figure>
```

Comment:
Why use figure if you're not including a caption? I think the `img` tag alone should work fine in that case.